File: leetcode/python/226_invert_binary_tree.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def invertTree(self, root: TreeNode | None) -> TreeNode | None:
        if not root:
            return root
        stack = [root]
        while stack:
            p = stack.pop()
            p.left, p.right = p.right, p.left
            if p.left:
                stack.append(p.left)
            if p.right:
                stack.append(p.right)
        return root
        

File: leetcode/c++/226_invert_binary_tree.cpp
#include <vector>

using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        if (root == nullptr) return root;

        vector<TreeNode*> stack = {root};
        while (!stack.empty()) {
            TreeNode* node = stack.back();
            stack.pop_back();
            if (node == nullptr) continue;
            swap(node->left, node->right);
            stack.push_back(node->left);
            stack.push_back(node->right);
        }
        return root;
    }
};